### DISCUSSION TITLE
what is three table differences?

### DISCUSSION THREAD

**Discussion by xiaohu-zhang** [ByConity/ByConity#580] [Q&A] [answered]
1. CREATE TABLE bucket (name String, age Int64) ENGINE = CnchMergeTree() PARTITION BY name CLUSTER BY age INTO 1 BUCKETS ORDER BY name;
2. CREATE TABLE bucket_with_split_number (name String, age Int64) ENGINE = CnchMergeTree() PARTITION BY name CLUSTER BY (name, age) INTO 1 BUCKETS SPLIT_NUMBER 60 ORDER BY name;
3. CREATE TABLE dts_bucket_with_split_number_n_range (name String, age Int64) ENGINE = CnchMergeTree() PARTITION BY name CLUSTER BY (name) INTO 1 BUCKETS SPLIT_NUMBER 60 WITH_RANGE ORDER BY name;

what is no SPLIT_NUMBER , SPLIT_NUMBER and SPLIT_NUMBER WITH_RANGE difference? thank you

**Comment by hustnn** [comment, marked as answer]
I think you can only need to use CLUSTER BY xxx INTO # BUCKETS, the others are mainly for some internal use cases. https://byconity.github.io/zh-cn/docs/basic-guide/data-import#%E7%9B%B4%E6%8E%A5%E5%86%99%E5%85%A5%E6%96%B9%E5%BC%8F%E8%B0%83%E4%BC%98
